pkg/module_manager/scheduler/scheduler.go

```diff
@@ -0,0 +1,459 @@
+package scheduler
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/dominikbraun/graph"
+	"github.com/dominikbraun/graph/draw"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/flant/addon-operator/pkg/module_manager/scheduler/extenders"
+	"github.com/flant/addon-operator/pkg/module_manager/scheduler/node"
+)
+
+type Scheduler struct {
+	extenders []Extender
+
+	l                   sync.Mutex
+	dag                 graph.Graph[string, *node.Node]
+	root                *node.Node
+	enabledModules      *[]string
+	retrospectiveStatus map[extenders.ExtenderName]map[string]bool
+}
+
+type Extender interface {
+	Name() extenders.ExtenderName
+	Filter(module node.ModuleInterface) (*bool, error)
+	IsShutter() bool
+	IsResettable() bool
```

Comment:
Do we really need that?
We can just create an empty Reset function which do nothing
```
func (a SomeExtender) Reset() error {
  return nil
}
```

Reply:
yeah, my bad, fixed it